File: backend/src/Fix.Application/UseCases/Orders/Commands/UpdateOrderCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Orders.Dtos;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Orders.Commands;

[RequireRole(RoleCodes.UpdateOrder)]
public sealed record UpdateOrderCommand(Guid UserId, Guid OrganizationId, Guid Id, Guid CounterpartyId, OrderTermsInput Terms)
    : ICommand<OrderDto>, IOrganizationRequest;

File: frontend/bff/src/application/orders/commands/update-order.command.ts
import type { OrderTermsInput } from './order-terms.input.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface UpdateOrderCommand {
  context: RequestContext;
  id: string;
  counterpartyId: string;
  terms: OrderTermsInput;
}
